crates/iota-grpc-api/tests/checkpoint_e2e.rs

```diff
@@ -0,0 +1,193 @@
+    let (cluster, mut client) = setup_test_cluster_and_client().await;
+
+    let sender = cluster.get_address_0();
+    let receiver = cluster.get_address_1();
+
+    // Wait for 2 new checkpoint to be available
+    cluster.wait_for_checkpoint(2, None).await;
+
+    // Advance to a new epoch
+    cluster.force_new_epoch().await;
+    cluster.transfer_iota_must_exceed(sender, receiver, 1).await;
+
+    // Wait for 3 new checkpoints in the new epoch
+    cluster.wait_for_checkpoint(3, None).await;
+    cluster.force_new_epoch().await;
+    cluster.transfer_iota_must_exceed(sender, receiver, 1).await;
+
+    // List all checkpoints and their epochs using the gRPC stream
+    let mut stream = client
+        .stream_checkpoints(Some(0), None, false)
+        .await
+        .expect("gRPC stream");
+    let mut all_indices = vec![];
+    let mut all_epochs = vec![];
+
+    tokio::time::timeout(Duration::from_secs(120), async {
+        while let Some(res) = stream.next().await {
+            match res {
+                Ok(checkpoint_content) => match checkpoint_content {
+                    iota_grpc_api::client::CheckpointContent::Summary(summary) => match summary {
+                        iota_grpc_types::CertifiedCheckpointSummary::V1(v1_summary) => {
+                            let epoch = v1_summary.data().epoch;
+                            all_indices.push(v1_summary.data().sequence_number);
+                            all_epochs.push(epoch);
+                            if v1_summary.data().sequence_number > 50 {
+                                break;
+                            }
+                        }
+                    },
+                    iota_grpc_api::client::CheckpointContent::Data(_) => {
+                        panic!("Expected summary, got data");
+                    }
+                },
+                Err(e) => {
+                    panic!("gRPC stream error: {e:?}");
+                }
+            }
+        }
+    })
+    .await
+    .expect("waiting for checkpoints timed out");
+
+    // Query for the first checkpoint of epoch 0 (should be 0)
+    let first_0 = client
+        .get_epoch_first_checkpoint_sequence_number(0)
+        .await
+        .expect("gRPC call");
+    assert_eq!(first_0, 0, "First checkpoint of epoch 0 should be 0");
+
+    // Query for the first checkpoint of epoch 1 (should be >= 2)
+    let first_1 = client
+        .get_epoch_first_checkpoint_sequence_number(1)
+        .await
+        .expect("gRPC call");
+    assert!(
+        first_1 >= 2,
+        "First checkpoint of epoch 1 should be >= 2, got {first_1}"
+    );
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_stream_full_checkpoint_data() {
+    let (cluster, mut client) = setup_test_cluster_and_client().await;
+
+    //  Ensures the checkpoint executor is properly initialized and running before
+    // the gRPC streaming
+    cluster.wait_for_checkpoint(0, None).await;
```

Comment:
Doesn't look like a proper solution: client can cause server to panic. This should be fixed.